files/image_config/monit/memory_checker

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+import argparse
+import subprocess
+import sys
+import syslog
+import re
+
+
+def get_command_result(command):
+    """Executes the command and return the resulting output.
+
+    Args:
+        command: A string contains the command to be executed.
+
+    Returns:
+        A string which contains the output of command.
+    """
+    command_stdout = ""
+
+    try:
+        proc_instance = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                         shell=True, universal_newlines=True)
+        command_stdout, command_stderr = proc_instance.communicate()
+        if proc_instance.returncode != 0:
+            syslog.syslog(syslog.LOG_ERR, "[memory_checker] Failed to execute the command '{}'. Return code: '{}'"
+                          .format(command, proc_instance.returncode))
+            sys.exit(1)
+    except (OSError, ValueError) as err:
+        syslog.syslog(syslog.LOG_ERR, "[memory_checker] Failed to execute the command '{}'. Error: '{}'"
+                      .format(command, err))
+        sys.exit(2)
+
+    return command_stdout.strip()
+
+
+def check_memory_usage(container_name, threshold_value):
+    """Checks the memory usage of a container and writes an alerting messages into
+    the syslog if the memory usage is larger than the threshold value.
+
+    Args:
+        container_name: A string represtents name of a container
+        threshold_value: An integer indicates the threshold value (Bytes) of memory usage.
+
+    Returns:
+        None.
+    """
+    command = "docker stats --no-stream --format \{{\{{.MemUsage\}}\}} {}".format(container_name)
+    command_stdout = get_command_result(command)
+    mem_usage = command_stdout.split("/")[0].strip()
+    match_obj = re.match(r"\d+\.?\d*", mem_usage)
+    if match_obj:
+        mem_usage_value = float(mem_usage[match_obj.start():match_obj.end()])
+        mem_usage_unit = mem_usage[match_obj.end():]
```

Comment:
@jleveque At here, I use regex to retrieve the memory usage value and memory usage unit from the content of variable `mem_usage`. For example, if the content of variable `mem_usage` is `0B`, then `mem_usage_value` will be `0` and `mem_usage_unit` will be `B`. 

Can you please help me review whether there is a better solution for this?

Reply:
Maybe you could do what procdockerstatsd does?

https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-host-services/scripts/procdockerstatsd#L40-L50

In the future we could make this common code and share it.